src/v/http/client.cc

```diff
@@ -44,6 +44,11 @@ namespace http {
 
 // client implementation //
 
+// Close all connections that were used more than 5 seconds ago.
```

Comment:
> The 5s timeout is hardcoded. AWS S3 uses 10s timeout but other cloud
storage providers might be different. The 5s timeout is small enough

Can we imagine scenarios where we might need to change this or is hard coding 5s really fine?